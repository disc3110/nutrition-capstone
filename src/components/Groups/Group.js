import { NavLink } from 'react-router-dom';

const Group = (props) => {
  const { name, number, slug } = props

  return (
    <NavLink className="links p-3 min-h-5 group" to={`/sports/${slug}`} >
        <h2 className="text-end no-select fw-bold">{name.toUpperCase()}</h2>
        <p className="text-end no-select">{`${number} sports`}</p>
    </NavLink>
  )
}

export default Group