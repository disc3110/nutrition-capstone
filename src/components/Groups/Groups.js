import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchGroups } from '../../Redux/Sports/groups';
import { fetchSports } from '../../Redux/Sports/sportsList';
import Group from './Group';
import '../../style/groups.css';

const Groups = () => {
  const dispatch = useDispatch();
  const groups = useSelector((state) => state.groups.entities);
  const sports = useSelector((state) => state.sports.entities);

  useEffect(() => {
    if (groups.length === 0) {
      dispatch(fetchGroups());
    }
    if (sports.length === 0) {
      dispatch(fetchSports());
    }
  });

  const renderGroups = (groupCollection) => groupCollection.map((groupInfo) => (
    <Group
      key={groupInfo.id}
      name={groupInfo.attributes.name}
      number={groupInfo.relationships.sports.data.length}
      slug={groupInfo.attributes.slug}
    />
  ));

  return (
    <div className="d-flex flex-wrap group-container">{renderGroups(groups)}</div>
  );
};

export default Groups;
