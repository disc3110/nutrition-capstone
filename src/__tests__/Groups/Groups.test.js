import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Groups from '../../components/Groups/Groups';
import store from '../../Redux/configureStore';

describe('Check if all components are rendered', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Groups /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
