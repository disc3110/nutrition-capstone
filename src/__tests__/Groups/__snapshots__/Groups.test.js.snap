// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if all components are rendered renders correctly 1`] = `
<div
  className="d-flex flex-wrap group-container"
/>
`;
